tailfs/local.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package tailfs
+
+import (
+	"log"
+	"net"
+	"net/http"
+
+	"github.com/tailscale/xnet/webdav"
+	"tailscale.com/connlistener"
+	"tailscale.com/tailfs/compositefs"
+	"tailscale.com/tailfs/webdavfs"
+	"tailscale.com/types/logger"
+)
+
+// Remote represents a remote TailFS node.
+type Remote struct {
+	Name      string
+	URL       string
+	Available func() bool
+}
+
+// ForLocal is the TailFS filesystem exposed to local clients. It provides a
+// unified WebDAV interface to remote TailFS shares on other nodes.
+type ForLocal interface {
```

Comment:
As noted elsewhere, I don't think this needs to be an interface, we can make this the concrete implementation.

Reply:
As noted on the note elsewhere, I prefer to avoid returning concrete implementations of services.